### ISSUE TITLE
Qupzilla core dump

### ISSUE THREAD

**Issue by stefonarch** [QupZilla/qupzilla#2150]
Qupzilla crashing always when trying to restore session, seen on two machines 32 bit with arch linux with recent update, probably
[2016-12-28 09:00] upgraded qt5-webengine (5.7.0-7 -> 5.7.1-1)

Delete profile (only until next :restore) or private browsing works.
Similar block and crashes can bee seen loading  sites like www.blitzortung.org which uses  flash.

```
$ qupzilla
QupZilla: 5 extensions loaded
QupZilla: Crashed :( Saving backtrace in /home/stef/.config/qupzilla/crashlog ...
Backtrace successfully saved in /home/stef/.config/qupzilla/crashlog/Crash-2016-12-28T18:43:10.txt
Annullato (core dump creato)
Time: mer dic 28 18:41:38 2016
Qt version: 5.7.1 (compiled with 5.7.0)
QupZilla version: 2.0.2
Rendering engine:  QtWebEngine

> ==============BACKTRACE==============
#0: qupzilla(_Z23qupzilla_signal_handleri+0x1ae) [0x804aebe]
#1: linux-gate.so.1(__kernel_sigreturn+0) [0xb778dcf0]
#2: /usr/lib/libQt5Core.so.5(_ZNK10QJsonArray13toVariantListEv+0x34) [0xb1410c94]
#3: /usr/lib/libQt5Core.so.5(+0x2af5f8) [0xb147a5f8]
#4: /usr/lib/libQt5Core.so.5(_ZNK8QVariant6toListEv+0xb4) [0xb14797d4]
#5: /usr/lib/libQt5WebChannel.so.5(_ZN20QMetaObjectPublisher10wrapResultERK8QVariantP28QWebChannelAbstractTransportRK7QString+0xa60) [0xb0657fb0]
#6: /usr/lib/libQt5WebChannel.so.5(_ZN20QMetaObjectPublisher13handleMessageERK11QJsonObjectP28QWebChannelAbstractTransport+0x87f) [0xb065b5df]
#7: /usr/lib/libQt5Core.so.5(_ZN11QMetaObject8activateEP7QObjectiiPPv+0x82e) [0xb145d7ce]
#8: /usr/lib/libQt5Core.so.5(_ZN11QMetaObject8activateEP7QObjectPKS_iPPv+0x2d) [0xb145dcad]
#9: /usr/lib/libQt5WebChannel.so.5(_ZN28QWebChannelAbstractTransport15messageReceivedERK11QJsonObjectPS_+0x51) [0xb0661491]
#10: /usr/lib/libQt5WebEngineCore.so.5(+0x279172) [0xb2035172]
#11: /usr/lib/libQt5WebEngineCore.so.5(+0x2793de) [0xb20353de]
#12: /usr/lib/libQt5WebEngineCore.so.5(+0xfa69d1) [0xb2d629d1]
#13: /usr/lib/libQt5WebEngineCore.so.5(+0xfa83e6) [0xb2d643e6]
#14: /usr/lib/libQt5WebEngineCore.so.5(+0xf14cde) [0xb2cd0cde]
#15: /usr/lib/libQt5WebEngineCore.so.5(+0xf2393d) [0xb2cdf93d]
#16: /usr/lib/libQt5WebEngineCore.so.5(+0xf07cc3) [0xb2cc3cc3]
#17: /usr/lib/libQt5WebEngineCore.so.5(+0x18b8eba) [0xb3674eba]
#18: /usr/lib/libQt5WebEngineCore.so.5(+0x18b8888) [0xb3674888]
#19: /usr/lib/libQt5WebEngineCore.so.5(+0x1cb13c6) [0xb3a6d3c6]
#20: /usr/lib/libQt5WebEngineCore.so.5(+0x1c544b2) [0xb3a104b2]
#21: /usr/lib/libQt5WebEngineCore.so.5(+0x1c5511a) [0xb3a1111a]
#22: /usr/lib/libQt5WebEngineCore.so.5(+0x1c55621) [0xb3a11621]
#23: /usr/lib/libQt5WebEngineCore.so.5(+0x23623b) [0xb1ff223b]
#24: /usr/lib/libQt5Core.so.5(_ZN7QObject5eventEP6QEvent+0x96) [0xb145e206]
#25: /usr/lib/libQt5Widgets.so.5(_ZN19QApplicationPrivate13notify_helperEP7QObjectP6QEvent+0x9a) [0xb179112a]
#26: /usr/lib/libQt5Widgets.so.5(_ZN12QApplication6notifyEP7QObjectP6QEvent+0x27c) [0xb1798fbc]
#27: /usr/lib/libQt5Core.so.5(_ZN16QCoreApplication15notifyInternal2EP7QObjectP6QEvent+0x12d) [0xb142fb5d]
#28: /usr/lib/libQt5Core.so.5(_ZN23QCoreApplicationPrivate16sendPostedEventsEP7QObjectiP11QThreadData+0x2de) [0xb143256e]
#29: /usr/lib/libQt5Core.so.5(_ZN16QCoreApplication16sendPostedEventsEP7QObjecti+0x27) [0xb14329c7]
#30: /usr/lib/libQt5Core.so.5(+0x2bdd83) [0xb1488d83]
#31: /usr/lib/libglib-2.0.so.0(g_main_context_dispatch+0x299) [0xaf9bc2c9]
#32: /usr/lib/libglib-2.0.so.0(+0x47569) [0xaf9bc569]
#33: /usr/lib/libglib-2.0.so.0(g_main_context_iteration+0x34) [0xaf9bc634]
#34: /usr/lib/libQt5Core.so.5(_ZN20QEventDispatcherGlib13processEventsE6QFlagsIN10QEventLoop17ProcessEventsFlagEE+0x63) [0xb1489173]
#35: /usr/lib/libQt5XcbQpa.so.5(+0x656b1) [0xa9d9d6b1]
#36: /usr/lib/libQt5Core.so.5(_ZN10QEventLoop4execE6QFlagsINS_17ProcessEventsFlagEE+0x11d) [0xb142dc1d]
#37: /usr/lib/libQt5Core.so.5(_ZN16QCoreApplication4execEv+0x9f) [0xb1436b6f]
#38: /usr/lib/libQt5Gui.so.5(_ZN15QGuiApplication4execEv+0x21) [0xb072bb31]
#39: /usr/lib/libQt5Widgets.so.5(_ZN12QApplication4execEv+0x14) [0xb1791084]
#40: qupzilla() [0x804a2bc]
#41: /usr/lib/libc.so.6(__libc_start_main+0xf6) [0xb0e8c196]
#42: qupzilla() [0x804a724]
```

**Comment by DebugReport** [comment]
Crashed sometimes.

> Received signal 11 SEGV_MAPERR 000000000000
#0 0x7fbd6e49b97e <unknown>
#1 0x7fbd6e49bd59 <unknown>
#2 0x7fbd6b84a0b0 <unknown>
#3 0x7fbd6d493876 <unknown>
#4 0x7fbd6f0d3e0e <unknown>
#5 0x7fbd6f0baeb3 <unknown>
#6 0x7fbd6f079984 <unknown>
#7 0x7fbd6f079acf <unknown>
#8 0x7fbd6f71b9ba <unknown>
#9 0x7fbd6fd88f14 <unknown>
#10 0x7fbd6fd89039 <unknown>
#11 0x7fbd6fd8b7f2 <unknown>
#12 0x7fbd6fd43cec <unknown>
#13 0x7fbd6f66c98d <unknown>
#14 0x7fbd6f72b370 <unknown>
#15 0x7fbd6f72b55e <unknown>
#16 0x7fbd6f72b6de <unknown>
#17 0x7fbd6f72b753 <unknown>
#18 0x7fbd6f72b7ed <unknown>
#19 0x7fbd6f60e78f <unknown>
#20 0x7fbd6fa258bf <unknown>
#21 0x7fbd6fa258fe <unknown>
#22 0x7fbd6cdf102c <unknown>
#23 0x7fbd6cdf34de <unknown>
#24 0x7fbd6cdf4b8b <unknown>
#25 0x7fbd6ed0efe0 <unknown>
#26 0x7fbd6e5128bd <unknown>
#27 0x7fbd6ed16184 <unknown>
#28 0x7fbd6ed167b7 <unknown>
#29 0x7fbd6e5128bd <unknown>
#30 0x7fbd6e4b915d <unknown>
#31 0x7fbd6e4b9d6d <unknown>
#32 0x7fbd6e4ba058 <unknown>
#33 0x7fbd6e4bbb19 <unknown>
#34 0x7fbd6e4d1cd4 <unknown>
#35 0x7fbd6e4b7f35 <unknown>
#36 0x7fbd6e2f4720 <unknown>
#37 0x7fbd6d0303fe <unknown>
#38 0x7fbd6d030908 <unknown>
#39 0x7fbd6d02eb89 <unknown>
#40 0x7fbd6c92f7a4 QtWebEngine::processMain()
#41 0x0000004008e3 <unknown>
#42 0x7fbd6b837291 __libc_start_main
#43 0x00000040093a <unknown>
  r8: 0000391174a04399  r9: 0000273feae18231 r10: 0000000000000002 r11: 0000000000000000
 r12: 00007fff5c57d0e0 r13: 0000000000000000 r14: 0000000000000004 r15: 000000000231e870
  di: 00000000022bedf0  si: 0000273feae2ffb9  bp: 000000000231e858  bx: 00000000022c0b40
  dx: 000000000234a270  ax: 00007fbd71c852cc  cx: 0000000000000002  sp: 00007fff5c57d0a0
  ip: 00007fbd6d493876 efl: 0000000000010246 cgf: 002b000000000033 erf: 0000000000000004
 trp: 000000000000000e msk: 0000000000000000 cr2: 0000000000000000
[end of stack trace]
QupZilla: Crashed :( Saving backtrace in /home/sorcus/.config/qupzilla/crashlog ...
Backtrace successfully saved in /home/sorcus/.config/qupzilla/crashlog/Crash-2017-01-18T14:04:53.txt
[26995:26995:0100/000000:ERROR:broker_posix.cc(41)] Invalid node channel message
zsh: abort (core dumped)  qupzilla

[Crash-2017-01-18T14:04:53.txt](https://github.com/QupZilla/qupzilla/files/714059/Crash-2017-01-18T14.04.53.txt)

**Comment by nowrep** [comment]
Looks same as #2151

**Comment by stefonarch** [comment]
Not exactly the same but related I, just tested it on my 64bit arch installation, it crashes with cltr+u and close but if I select "new session" qupzilla works fine again, but here there is no way reopening qupzilla without deleting the profile or using private navigation.

When killing qupzilla the restore is working (actually it does not display items) on 64bit and here with 32bit version it's not.

I compiled from AUR  and git version  is not affected.

**Comment by stefonarch** [comment]
Superseeded now by qupzilla version 2.1.0-1 and qwebengine 5.8.0-1

**Comment by nowrep** [comment]
This should now be fixed in QtWebEngine 5.8.

**Comment by stefonarch** [comment]
Confirmed as above ;)
